### PR TITLE
support data analysis engine (DAE)

### PR DESCRIPTION
**Pull request by tq444**
Weili Qian (1):
  drv/dae: support hashagg algorithm

Wenkai Lin (1):
  uadk: support for the DAE algorithm layer

 Makefile.am              |   28 +-
 drv/hisi_dae.c           | 1686 ++++++++++++++++++++++++++++++++++++++
 drv/hisi_qm_udrv.h       |    1 +
 include/drv/wd_agg_drv.h |   56 ++
 include/wd_agg.h         |  233 ++++++
 include/wd_alg.h         |    3 +
 include/wd_dae.h         |   80 ++
 include/wd_util.h        |    1 +
 libwd_dae.map            |   21 +
 wd_agg.c                 | 1572 +++++++++++++++++++++++++++++++++++
 wd_util.c                |    2 +
 11 files changed, 3680 insertions(+), 3 deletions(-)
 create mode 100644 drv/hisi_dae.c
 create mode 100644 include/drv/wd_agg_drv.h
 create mode 100644 include/wd_agg.h
 create mode 100644 include/wd_dae.h
 create mode 100644 libwd_dae.map
 create mode 100644 wd_agg.c

-- 
2.33.0

### CODE REVIEW

**Comment by qwl731** [comment]
+1

**Comment by lin755** [comment]
lgtm

File: include/wd_alg.h [hunk 1]
```diff
@@ -111,6 +111,7 @@ struct wd_alg_driver {
 	int (*send)(struct wd_alg_driver *drv, handle_t ctx, void *drv_msg);
 	int (*recv)(struct wd_alg_driver *drv, handle_t ctx, void *drv_msg);
 	int (*get_usage)(void *param);
+	int (*get_extend_ops)(void *ops);
```
Comment (Liulongfang) [inline include/wd_alg.h:114]: 这个修改需要在comment上说明清楚

File: drv/hisi_qm_udrv.h [hunk 2]
```diff
@@ -86,6 +86,7 @@ struct hisi_qp {
 	struct hisi_qm_queue_info q_info;
 	handle_t h_sgl_pool;
 	handle_t h_ctx;
+	void *priv;
```
Comment (Liulongfang) [inline drv/hisi_qm_udrv.h:89]: 这里添加一个priv是为了实现什么功能？有说明comment吗？

Comment (tq444) [reply drv/hisi_qm_udrv.h:90]: > 这里添加一个priv是为了实现什么功能？有说明comment吗？

已修改

**Review by Liulongfang — APPROVED** [review]
looks good to me

**Pull request merged**